src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -192,6 +196,33 @@ public async Task UpdateStateAsync(IProgress<IItemLoaderState> progress, Cancell
             NuGetEventTrigger.Instance.TriggerEvent(NuGetEvent.PackageLoadEnd);
         }
 
+        private async Task<SearchResult<IPackageSearchMetadata>> CombineSearchAsync(
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // get the browse/search results
+            SearchResult<IPackageSearchMetadata> browseResult = await _packageFeed.SearchAsync(_searchText, SearchFilter, cancellationToken);
+
+            // get the recommender results
+            SearchResult<IPackageSearchMetadata> recommenderResult = null;
+            if (_recommenderPackageFeed != null)
+            {
+                recommenderResult = await _recommenderPackageFeed.SearchAsync(_searchText, SearchFilter, cancellationToken);
+                _recommendedCount = recommenderResult.Count();
+            }
+
+            // update items list to include recommender results if needed
+            SearchResult<IPackageSearchMetadata> aggregated = browseResult;
+            if (recommenderResult != null)
+            {
+                IEnumerable<IPackageSearchMetadata> items = recommenderResult.Items.Concat(browseResult.Items);
```

Comment:
We decided to filter out duplicates if they appear in the first page (25 results). For example if we recommend Newtonsoft.Json, we would show it with the star, but then remove it from the non-starred list. For anything past the first 25 results, we'll just list duplicates so that we don't need to keep checking against the recomended list.